test/test_module_init.py

```diff
@@ -0,0 +1,330 @@
+        torch.nn.PReLU: ([], {}),
+        torch.nn.PairwiseDistance: ([], {}),
+        torch.nn.ParameterDict: ([], {}),
+        torch.nn.ParameterList: ([], {}),
+        torch.nn.PixelShuffle: ([2], {}),
+        torch.nn.PixelUnshuffle: ([2], {}),
+        torch.nn.PoissonNLLLoss: ([], {}),
+        torch.nn.RNN: ([5, 10], {}),
+        torch.nn.RNNBase: (['LSTM', 5, 10], {}),
+        torch.nn.RNNCell: ([5, 10], {}),
+        torch.nn.RNNCellBase: ([5, 10, True, 2], {}),
+        torch.nn.RReLU: ([], {}),
+        torch.nn.ReLU6: ([], {}),
+        torch.nn.ReLU: ([], {}),
+        torch.nn.ReflectionPad1d: ([2], {}),
+        torch.nn.ReflectionPad2d: ([2], {}),
+        torch.nn.ReplicationPad1d: ([2], {}),
+        torch.nn.ReplicationPad2d: ([2], {}),
+        torch.nn.ReplicationPad3d: ([2], {}),
+        torch.nn.SELU: ([], {}),
+        torch.nn.Sequential: ([], {}),
+        torch.nn.SiLU: ([], {}),
+        torch.nn.Sigmoid: ([], {}),
+        torch.nn.SmoothL1Loss: ([], {}),
+        torch.nn.SoftMarginLoss: ([], {}),
+        torch.nn.Softmax2d: ([], {}),
+        torch.nn.Softmax: ([], {}),
+        torch.nn.Softmin: ([], {}),
+        torch.nn.Softplus: ([], {}),
+        torch.nn.Softshrink: ([], {}),
+        torch.nn.Softsign: ([], {}),
+        torch.nn.SyncBatchNorm: ([5], {}),
+        torch.nn.Tanh: ([], {}),
+        torch.nn.Tanhshrink: ([], {}),
+        torch.nn.Threshold: ([0.1, 20], {}),
+        torch.nn.Transformer: ([], {}),
+        torch.nn.TransformerDecoder: ([torch.nn.TransformerDecoderLayer, 3], {}),
+        torch.nn.TransformerDecoderLayer: ([10, 2], {}),
+        torch.nn.TransformerEncoder: ([torch.nn.TransformerEncoderLayer, 3], {}),
+        torch.nn.TransformerEncoderLayer: ([10, 2], {}),
+        torch.nn.TripletMarginLoss: ([], {}),
+        torch.nn.TripletMarginWithDistanceLoss: ([], {}),
+        torch.nn.Unflatten: ([1, (2, 5, 5)], {}),
+        torch.nn.Unfold: ([3], {}),
+        torch.nn.Upsample: ([], {}),
+        torch.nn.UpsamplingBilinear2d: ([], {}),
+        torch.nn.UpsamplingNearest2d: ([], {}),
+        torch.nn.ZeroPad2d: ([0], {}),
+        torch.nn.qat.Conv2d: ([3, 3, 3], {'qconfig': torch.quantization.default_qconfig}),
+        torch.nn.qat.Conv3d: ([3, 3, 3], {'qconfig': torch.quantization.default_qconfig}),
+        torch.nn.qat.Linear: ([5, 2], {'qconfig': torch.quantization.default_qconfig}),
+        torch.nn.quantizable.LSTM: ([5, 6], {}),
+        torch.nn.quantizable.LSTMCell: ([5, 6], {}),
+        torch.nn.quantizable.MultiheadAttention: ([10, 2], {}),
+    }
+
+
+# Instantiates the given class with the given args, kwargs, optionally on a given device.
+def instantiate_class(cls, args, kwargs, device=None):
+    return cls(*args, **kwargs) if device is None else cls(*args, **kwargs, device=device)
+
+
+# Returns a function that calls the real implementation of a method
+# in addition to passing args to a mock object.
+def mock_wrapper(method):
```

Comment:
Use wraps (https://docs.python.org/3/library/functools.html#functools.wraps)

Reply:
I couldn't get `wraps` to work for this specific use case where I don't have a particular object. This wrapper is used below to hook into all parameter creations or buffer registrations. If you know how to use `wraps` for this, let me know and I'll change it.

Reply:
Interesting. I suppose this can wait until it's an issue (and it may never be an issue).